src/m365/planner/commands/plan/plan-details-get.ts

```diff
@@ -0,0 +1,182 @@
+  private groupId: string = "";
+
+  public get name(): string {
+    return commands.PLAN_DETAILS_GET;
+  }
+
+  public get description(): string {
+    return 'Get details of a Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.planId = typeof args.options.planId !== 'undefined';
+    telemetryProps.planTitle = typeof args.options.planTitle !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+  
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getGroupId(args)
+      .then((groupId: string): Promise<string> => {
+        this.groupId = groupId;
+        return this.getPlanId(args);
+      })
+      .then((planId: string): Promise<PlannerPlanDetails> => {
+        args.options.planId = planId;
+        return this.getPlanDetails(args);
+      })
+      .then((res: PlannerPlanDetails): void => {
+        logger.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getGroupId(args: CommandArgs): Promise<string> {
+    if (args.options.planId) {
+      return Promise.resolve("");
+    }
+
+    if (args.options.ownerGroupId) {
+      return Promise.resolve(args.options.ownerGroupId);
+    }
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups?$filter=displayName eq '${encodeURIComponent(args.options.ownerGroupName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: [{ id: string, resourceProvisioningOptions: string[] }] }>(requestOptions)
+      .then(response => {
+        const groupItem: { id: string, resourceProvisioningOptions: string[] } | undefined = response.value[0];
+
+        if (!groupItem) {
+          return Promise.reject(`The specified ownerGroup does not exist`);
+        }
+
+        if (response.value.length > 1) {
+          return Promise.reject(`Multiple ownerGroups with name ${args.options.ownerGroupName} found: Please choose between the following IDs ${response.value.map(x => x.id)}`);
+        }
+
+        return Promise.resolve(groupItem.id);
+      });
+  }
+
+  private getPlanId(args: CommandArgs): Promise<string> {
+    if (args.options.planId) {
+      return Promise.resolve(args.options.planId);
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups/${this.groupId}/planner/plans`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+    
+    return request
+      .get<{ value: PlannerPlan [] }>(requestOptions)
```

Comment:
If the specific Group would have many plans, this request would only return the first page. Instead, we should use the `this.getAllItems` method. For this we'd need to change the command's base class to `GraphItemsListCommand`.